### PR TITLE
fix: Updating endpoint to search by txHash and domain ID

### LINKED ISSUE #115: Update endpoint to search by txHash adding chainId as optional parameter
Since there is the chance -small one-, that txHash could not be unique, we should fetch transfer passing txHash + chainId

## Implementation details

- update current endpoint to receive chainId as optional parameter

## Testing details

- update test to reflect the change

## Acceptance Criteria

- tests are passing
- we are able to fetch transfer by txHash + chainId

### PR DESCRIPTION
**Pull request by mj52951**
Updated endpoint to search by txHash and domain ID. Also, updated affected tests and added schema for the mentioned route. 

closes #115

### CODE REVIEW

**Comment by wainola** [comment]
@MakMuftic just to confirm here, `domainId` is going to be a mandatory parameter, isn't it?

**Review by tcar121293 — APPROVED** [review]

**Comment by tcar121293** [comment]
> @MakMuftic just to confirm here, `domainId` is going to be a mandatory parameter, isn't it?

i would say yes

**Review by wainola — APPROVED** [review]

**Comment by mpetrun5** [comment]
> > @MakMuftic just to confirm here, `domainId` is going to be a mandatory parameter, isn't it?
> 
> i would say yes

Arguably, this is a downgrade as we have to then asks users for both the domain and the txHash when the user experience is better without that.

**Comment by tcar121293** [comment]
> > > @MakMuftic just to confirm here, `domainId` is going to be a mandatory parameter, isn't it?
> > 
> > 
> > i would say yes
> 
> Arguably, this is a downgrade as we have to then asks users for both the domain and the txHash when the user experience is better without that.
Do you suggest then that we put the domainID in qs? like /transfers/txHash/:txHash?domainID=1

File: src/routes.ts [hunk 1]
```diff
@@ -25,13 +27,15 @@ export async function routes(fastify: FastifyInstance): Promise<void> {
...
     url: "/transfers/:id",
+    schema: transferByIdSchema,
     handler: TransfersController.transferById,
   })
 
   fastify.route({
     method: "GET",
-    url: "/transfers/txHash/:txHash",
-    handler: TransfersController.transferByTxHash,
+    url: "/transfers/txHash/:txHash/domains/:domainID",
```
Comment (tcar121293) [inline src/routes.ts:36]: Considering Matija consern, let's go with `/transfers/txHash/:txHash?domainID=${domainID}` In that way we won't break the current implementation of explorer-ui, and we give possibility to filter it also by domainID @mj52951 @wainola

**Review by tcar121293 — CHANGES_REQUESTED** [review]

File: src/routes.ts (hunk 1 above)
Comment (wainola) [reply src/routes.ts]: Agree, will update the SDK PR that is going to use this route

**Review by wainola — APPROVED** [review]

**Review by MakMuftic — APPROVED** [review]

**Pull request merged**